File: dev/20190822/k2works/src/Application/Service/FizzBuzzService.ts
import { IFizzBuzz } from "../../Domain/Model/IFizzBuzz";
import { FizzBuzzType } from "../../Domain/Type/FizzBuzzType";
import { IFizzBuzzType } from "../../Domain/Type/IFizzBuzzType";
import { FizzBuzzListCommand } from "../Command/FizzBuzzListCommand";
import { FizzBuzzValueCommand } from "../Command/FizzBuzzValueCommand";
import { IFizzBuzzCommand } from "../Command/IFizzBuzzCommand";

export class FizzBuzzService {
  public generateList(type: string, input: number): IFizzBuzz {
    const command: IFizzBuzzCommand = new FizzBuzzListCommand(
      FizzBuzzType.valueOf(type)
    );
    return command.execute(input);
  }

  public generate(type: string, input: number): IFizzBuzz {
    const command: IFizzBuzzCommand = new FizzBuzzValueCommand(
      FizzBuzzType.valueOf(type)
    );
    return command.execute(input);
  }
}

File: dev/20190902/k2works/app/FizzBuzz/Application/Service/FizzBuzzService.cs
namespace FizzBuzz.Application.Service
{
    public class FizzBuzzService
    {
        private IFizzBuzzCommand _command;

        public FizzBuzzService(IFizzBuzzCommand command)
        {
            _command = command;
        }

        public IFizzBuzz Generate(int number)
        {
            return _command.Execute(number);
        }

        public IFizzBuzz GenerateList(int number)
        {
            return _command.Execute(number);
        }

    }
}